src/main/java/com/jcabi/aether/RepositoryAuthentication.java

```diff
@@ -67,11 +67,16 @@
      * Creates a new authentication with the specified properties.
      * @param auth The authentication object.
      */
+    @SuppressWarnings("PMD.ConfusingTernary")
     public RepositoryAuthentication(final Authentication auth) {
         this.username = auth.getUsername();
-        this.password = auth.getPassword().toCharArray();
+        // @checkstyle AvoidInlineConditionalsCheck (2 lines)
+        this.password = auth.getPassword() != null
```

Comment:
@pinaf Yes, it is required. All the fields of the [Authentication](http://sonatype.github.io/sonatype-aether/apidocs/org/sonatype/aether/repository/Authentication.html) might be null, therefore such a check is required. In maven proxy settings it is even impossible to provide `privatekeyfile` or `passphrase`.